File: sources/java/01_branches/Main.java
// task 01 branches — expected output: 33333334 13333333 7619048 45714285
// build: javac Main.java    run: java Main
// build (graalvm native-image): native-image -O2 Main    run: ./Main

public class Main {
    public static void main(String[] args) {
        long a = 0, b = 0, c = 0, d = 0;
        for (long i = 0; i < 100000000L; i++) {
            if (i % 3 == 0) {
                a += 1;
            } else if (i % 5 == 0) {
                b += 1;
            } else if (i % 7 == 0) {
                c += 1;
            } else {
                d += 1;
            }
        }
        System.out.println(a + " " + b + " " + c + " " + d);
    }
}

File: sources/kotlin/01_branches/main.kt
// task 01 branches — expected output: 33333334 13333333 7619048 45714285
// build: kotlinc main.kt -include-runtime -d prog.jar    run: java -jar prog.jar    [native build: kotlinc-native -opt -o prog main.kt    native run: ./prog]

fun main() {
    var a = 0L
    var b = 0L
    var c = 0L
    var d = 0L
    for (i in 0L until 100000000L) {
        if (i % 3L == 0L) {
            a++
        } else if (i % 5L == 0L) {
            b++
        } else if (i % 7L == 0L) {
            c++
        } else {
            d++
        }
    }
    println("$a $b $c $d")
}
